driver/src/main/scala/org/reactivecouchbase/client/Read.scala

```diff
@@ -158,6 +160,31 @@ trait Read {
   }
 
   /**
+    *
+    * Fetch a optional document and set its expiry
+    *
+    * @param key the key of the document
+    * @param exp expiration of the doc
+    * @param bucket the bucket to use
+    * @param r Json reader
+    * @param ec ExecutionContext for async processing
+    * @tparam T type of the doc
+    * @return
+    */
+  def getAndTouch[T](key: String, exp: CouchbaseExpirationTiming)(implicit bucket: CouchbaseBucket, r: Reads[T], ec: ExecutionContext): Future[Option[T]] = {
+    Future( bucket.couchbaseClient.asyncGetAndTouch(key, exp).get().getValue ) map {
+      case doc: String => Some(r.reads(Json.parse(doc)))
+      case null => None
+      case _ if bucket.failWithNonStringDoc => throw new IllegalStateException(s"Document '$key' is not a string ...")
+      case _ if !bucket.failWithNonStringDoc => None
+    } map {
+      case Some(JsSuccess(value, _)) => Some(value)
+      case Some(JsError(errors)) if bucket.jsonStrictValidation => throw new JsonValidationException("Invalid JSON content", JsError.toFlatJson(errors))
```

Comment:
It seems "toFlatJson" has been deprecated and its replacement seems straightforward to add.